### PR TITLE
Import Quarkus BOM

### PR DESCRIPTION
**Pull request by rsynek**
The part we agreed on from #1438 , importing the Quarkus BOM without additional changes.

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] native</b>
</details>

### CODE REVIEW

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml [hunk 1]
```diff
@@ -171,6 +165,18 @@
...
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <configuration>
+          <ignoredUsedUndeclaredDependencies>
+            <dependency>org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec</dependency>
+          </ignoredUsedUndeclaredDependencies>
+          <ignoredUnusedDeclaredDependencies>
+            <dependency>jakarta.transaction:jakarta.transaction-api</dependency>
+          </ignoredUnusedDeclaredDependencies>
```
Comment (triceo) [inline optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:177]: I'm assuming that this is a runtime dependency and the plugin simply doesn't see that we will crash without it?

**Review by triceo — APPROVED** [review]
Sooo many removals! <3

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml [hunk 2]
```diff
@@ -171,6 +165,18 @@
       </testResource>
     </testResources>
     <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <configuration>
+          <ignoredUsedUndeclaredDependencies>
+            <dependency>org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec</dependency>
+          </ignoredUsedUndeclaredDependencies>
```
[the remark this answers is not in the document]
Comment (rsynek) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:174]: Now I see it's a bit more complicated: the `org.jboss.spec` one comes as a test dependency from `kie-test-util` and also as an optional runtime dependency from `hibernate-core`.

Option 1) exclude the `org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec` and use the `jakarta.transaction-api`, which the Quarkus BOM defines version for.

Option 2) stay with the `org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec` and manage its version ourselves.

[the remark this answers is not in the document]
Comment (rsynek) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:174]: I am in favor of Option 1) as sooner or later, the `hibernate-core` will switch to the new `jakarta-transaction-api`.

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml (hunk 1 above)
Comment (rsynek) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:177]: See my reply to your previous comment; once we solve it, this extra configuration might go.

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (triceo) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:174]: +1 on 1), provided we explain that in the POM.

**Pull request merged**